main/util/HullPolygon.ts

```diff
@@ -99,26 +109,31 @@ export class HullPolygon {
     const skipRight =
       lowerIndexes[lowerIndexes.length - 1] ===
       upperIndexes[upperIndexes.length - 1];
-    const hull: Polygon = [];
+    const hullPoints: Point[] = [];
 
     // Add upper hull in right-to-left order.
     // Then add lower hull in left-to-right order.
     for (i = upperIndexes.length - 1; i >= 0; --i)
-      hull.push(points[sortedPoints[upperIndexes[i]].index]);
+      hullPoints.push(pointArray[sortedPoints[upperIndexes[i]].index]);
     for (
       i = skipLeft ? 1 : 0;
       i < lowerIndexes.length - (skipRight ? 1 : 0);
       ++i
     )
-      hull.push(points[sortedPoints[lowerIndexes[i]].index]);
+      hullPoints.push(pointArray[sortedPoints[lowerIndexes[i]].index]);
 
-    return hull;
+    return new Polygon(hullPoints);
   }
 
   /**
    * Returns true if and only if the specified point is inside the specified polygon.
    */
-  public static contains(polygon: Polygon, point: Point): boolean {
+  public static contains(polygon: Polygon | PolygonArray, point: Point): boolean {
```

Comment:
I haven’t looked at the uses of this function but it seems like the member function on Polygon should just be called directly.